internal/testrunner/runners/pipeline/runner.go

```diff
@@ -0,0 +1,158 @@
+package pipeline
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/elasticsearch"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/testrunner"
+)
+
+const (
+	// TestType defining pipeline tests
+	TestType testrunner.TestType = "pipeline"
+)
+
+type runner struct {
+	options testrunner.TestOptions
+}
+
+// Run runs the pipeline tests defined under the given folder
+func Run(options testrunner.TestOptions) error {
+	r := runner{options}
+	return r.run()
+}
+
+func (r *runner) run() error {
+	testCaseFiles, err := r.listTestCaseFiles()
+	if err != nil {
+		return errors.Wrap(err, "listing test case definitions failed")
+	}
+
+	datasetPath, found, err := packages.FindDatasetRootForPath(r.options.TestFolderPath)
+	if err != nil {
+		return errors.Wrap(err, "locating dataset root failed")
+	}
+	if !found {
+		return errors.New("dataset root not found")
+	}
+
+	esClient, err := elasticsearch.Client()
+	if err != nil {
+		return errors.Wrap(err, "fetching Elasticsearch client instance failed")
+	}
+
+	entryPipeline, pipelineIDs, err := installIngestPipelines(esClient, datasetPath)
+	if err != nil {
+		return errors.Wrap(err, "installing ingest pipelines failed")
+	}
+	defer func() {
+		err := uninstallIngestPipelines(esClient, pipelineIDs)
+		if err != nil {
+			fmt.Printf("uninstalling ingest pipelines failed: %v", err)
+		}
+	}()
+
+	var failed bool
+	for _, testCaseFile := range testCaseFiles {
+		tc, err := r.loadTestCaseFile(testCaseFile)
+		if err != nil {
+			return errors.Wrap(err, "loading test case failed")
+		}
+		fmt.Printf("Test case: %s\n", tc.name)
```

Comment:
Should this use the logger?

Reply:
It depends which parts do we prefer to keep as default messages/logs. You can take a look at log dumps in the issue description. Do you see something that can be delegated to a logger?

Reply:
I see your point. There's logs to help understand what the command is doing and then there's the actual end result output of the command. I would consider this statement here part of the command's output so I'm good with keeping this as-is.